dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/JspOptions.java

```diff
@@ -1072,9 +1086,26 @@ public void setJdkSourceLevel(int jdkSourceLevel) {
         this.jdkSourceLevel = jdkSourceLevel;
     }
 
+    public void setJavaSourceLevel(int javaSourceLevel) {
+        int jmv = JavaInfo.majorVersion();
+        if (javaSourceLevel > jmv) {
+            // can not specify higher than running Java
+            if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.INFO)) {
+                logger.logp(Level.INFO, CLASS_NAME, "setJavaSourceLevel", "Requested javaSourceLevel=" + javaSourceLevel + 
+                            " exceeds installed Java " + jmv + ", therefore forcing to " + jmv);
```

Comment:
Should be a warning in order to be more visible.

Reply:
I think a warning is appropriate here. Our docs specify: `Alerts user to a condition that might cause problems in the future such as configuration issues, usage of default values, and environmental conditions` Since this is a configuration issue.